### PR TITLE
fix extending torch native API docs

### PR DESCRIPTION
**Pull request by pmeier**
Couldn't think of a better `release notes:` label. Feel free to set a more fitting one

### CODE REVIEW

File: docs/source/notes/extending.rst [hunk 1]
```diff
@@ -866,7 +866,7 @@ C++ level.
 
 To effectively use this feature, it is important to know how the native part of
 PyTorch is implemented. The most important component there is what we call the
-"dispatcher" (the best description can be found in this [blog post](http://blog.ezyang.com/2020/09/lets-talk-about-the-pytorch-dispatcher/) even though it is slightly outdated). As
+"dispatcher" (the best description can be found in this `blog post <http://blog.ezyang.com/2020/09/lets-talk-about-the-pytorch-dispatcher/>`_ even though it is slightly outdated). As
```
[the remark this answers is not in the document]
Comment (pmeier) [reply docs/source/notes/extending.rst:869]: With this PR:

image

File: docs/source/notes/extending.rst [hunk 2]
```diff
@@ -881,24 +881,26 @@ Finally, we reach the backend kernel for CUDA which will launch the right CUDA k
...
+One configuration of the dispatcher is the order in which all these feature and backend keys are called. The latest list and their order can be found in ``DispatchKey.h`` inside the ``DispatchKey`` enum. For the purpose of extending torch, the important subset of the ordering for this discussion is:
+
+vmap -> Autocast -> Autograd -> ZeroTensor -> Neg/Conj -> Functionalize -> Python -> Backends
+
+The most important key for the purpose of this discussion is ``Python`` as every Tensor subclass with the ``__torch_dispatch__`` method defined will call into this feature. It is from there that the user-defined method is called and where the behavior can be overwritten arbitrarily. From there, calling the provided ``func`` again will perform a "redispatch".
 
 Some important implications of this implementation are:
+
```
[the remark this answers is not in the document]
Comment (pmeier) [reply docs/source/notes/extending.rst:891]: With this PR:

image

**Review by mikaylagawarecki — APPROVED** [review]
Thanks!

**Pull request closed without merging**